gio-subclass/src/application.rs

```diff
@@ -0,0 +1,599 @@
+    //             .unwrap_or(())
+    //     }
+    // }
+
+    fn parent_run_mainloop(&self) {
+        unsafe {
+            let klass = self.get_class();
+            let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+            (*parent_klass)
+                .run_mainloop
+                .map(|f| f(self.to_glib_none().0))
+                .unwrap_or(())
+        }
+    }
+
+    fn parent_quit_mainloop(&self) {
+        unsafe {
+            let klass = self.get_class();
+            let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+            (*parent_klass)
+                .quit_mainloop
+                .map(|f| f(self.to_glib_none().0))
+                .unwrap_or(())
+        }
+    }
+
+    fn parent_shutdown(&self) {
+        unsafe {
+            let klass = self.get_class();
+            let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+            (*parent_klass)
+                .shutdown
+                .map(|f| f(self.to_glib_none().0))
+                .unwrap_or(())
+        }
+    }
+
+    //...
+
+
+    // fn parent_handle_local_options(&self, options: &glib::VariantDict) {
+    //     unsafe {
+    //         let klass = self.get_class();
+    //         let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+    //         (*parent_klass)
+    //             .handle_local_options
+    //             .map(|f| f(self.to_glib_none().0, options.to_glib_none().0))
+    //             .unwrap_or(())
+    //     }
+    // }
+
+
+}
+
+pub unsafe trait ApplicationClassExt<T: ApplicationBase>
+where
+    T::ImplType: ApplicationImpl<T>,
+{
+    fn override_vfuncs(&mut self, _: &ClassInitToken) {
+        unsafe {
+            let klass = &mut *(self as *const Self as *mut gio_ffi::GApplicationClass);
+            klass.startup = Some(application_startup::<T>);
+            klass.activate = Some(application_activate::<T>);
+            klass.open = Some(application_open::<T>);
+            klass.command_line = Some(application_command_line::<T>);
+            klass.local_command_line = Some(application_local_command_line::<T>);
+            klass.before_emit = Some(application_before_emit::<T>);
+            klass.after_emit = Some(application_after_emit::<T>);
+            // klass.add_platform_data = Some(application_add_platform_data::<T>);
+            klass.quit_mainloop = Some(application_quit_mainloop::<T>);
+            klass.run_mainloop = Some(application_run_mainloop::<T>);
+            klass.shutdown = Some(application_shutdown::<T>);
+
+            //...
+            // klass.handle_local_options = Some(application_handle_local_options::<T>);
+
+
+        }
+    }
+}
+
+glib_wrapper! {
+    pub struct Application(Object<InstanceStruct<Application>>):
+        [gio::Application => gio_ffi::GApplication];
```

Comment:
We would ideally also duplicate the interfaces here: https://github.com/gtk-rs/gio/blob/d4fc17aa32983620caa136e0851d9a8063668c5e/src/auto/application.rs#L32

Reply:
I know, but that got me a bit too far for this initial PR. I'll get to that later. My plan is to implement these classes as I need them. And I guess I'll need those interfaces pretty soon :)

Reply:
Oh I just meant listing them here, no need to implement wrappers for those interfaces yet. Just adding the two interfaces to the `[]` list of "parents" here in this line